tests/libvmi/cloudinit.go

```diff
@@ -22,6 +22,8 @@ package libvmi
 import (
 	"encoding/base64"
 
+	k8scorev1 "k8s.io/api/core/v1"
```

Comment:
must we add this dependency to libvmi? maybe you can keep `WithCloudInitNoCloudEncodedNetworkData` out for now? I'm not sure myself.

p.s you use a very rare name for the imported package.

Reply:
`WithCloudInitNoCloudNetworkDataSecret` introduce this dependency.
Unfortunately its not possible to avoid because the Volume API depndes on it (`volume.CloudInitNoCloud.NetworkDataSecretRef` is type of `k8s.io/api/core/v1`).

Reply:
I know... Hence I suggest to keep this function out of this package. It has one users, so it can be defined there. As I said, I'm not 100% sure myself.

Reply:
The dependency already exists transitively, as `kubevirt.io/api/core/v1` depends on it.

Reply:
> @EdDev please use your `lgtm` powers, then.

There are other opened comments  and the linter is failing.